powershell-adapter/psDscAdapter/psDscAdapter.psm1

```diff
@@ -177,7 +187,9 @@ function LoadPowerShellClassResourcesFromModule
         foreach ($nestedModule in $moduleInfo.NestedModules)
         {
             $resourcesOfNestedModules = LoadPowerShellClassResourcesFromModule $nestedModule
-            $Resources.AddRange($resourcesOfNestedModules)
+            if ($resourcesOfNestedModules) {
```

Comment:
In this case, if there isn't a RootModule, then an empty collection is returned and we add that range which doesn't do anything but use CPU.  Would it be better to check `if ($resourcesOfNestedModules.Count -gt 0)` or is there a case where it might be `$null`?

Reply:
1) `if ($resourcesOfNestedModules) ` checks for both conditions: `-ne $null` **and** `.Count -gt 0`.
2) yes, `LoadPowerShellClassResourcesFromModule` can return `null` (in error cases).